File: backend/models/participant.go
package models

import "time"

type Participant struct {
	Id        int       `db:"id" json:"id"`
	FullName  string    `db:"full_name" json:"fullName"`
	Cipher    string    `db:"cipher" json:"cipher"`
	SchoolId  int       `db:"school_id" json:"schoolId"`
	ClassName string    `db:"class_name" json:"className"`
	Tasks     []Task    `db:"-" json:"tasks"`
	CreatedAt time.Time `db:"created_at" json:"createdAt"`
}

File: frontend/src/types/participant.ts
import { type Task } from "./task";
import { type Timestamps } from "./timestamps";

export interface Participant extends Timestamps {
  id: number;
  fullName: string;
  cipher: number;
  organizationId: number;
  group: string;
  tasks: Task[];
  total: number;
  percent: number;
  rating: string;
  status: string;
}
